src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -17,18 +17,19 @@ namespace Microsoft.Extensions.DependencyInjection
     /// </summary>
     internal class ServiceProvider : IServiceProvider, IDisposable
     {
-        private readonly ServiceProvider _root;
         private readonly ServiceTable _table;
         private bool _disposeCalled;
-
-        private readonly Dictionary<IService, object> _resolvedServices = new Dictionary<IService, object>();
         private List<IDisposable> _transientDisposables;
 
+        internal ServiceProvider Root { get; }
```

Comment:
Do we generally use `internal` or `public` on classes that are already `internal`?